kratos/utilities/shifted_boundary_meshless_interface_utility.cpp

```diff
@@ -0,0 +1,1124 @@
+        mMLSExtensionOperatorOrder = ThisParameters["mls_extension_operator_order"].GetInt();
+
+        // If true, the basis is created such that it is conforming with the linear FE space of the surrogate boundary
+        mConformingBasis = ThisParameters["conforming_basis"].GetBool();
+
+        // If true, the basis is created such that the surrogate boundary gradient is kept
+        const std::string ext_op_type = ThisParameters["extension_operator_type"].GetString();
+        if (ext_op_type == "MLS") {
+            mExtensionOperator = ExtensionOperator::MLS;
+        } else if (ext_op_type == "RBF") {
+            mExtensionOperator = ExtensionOperator::RBF;
+        } else if (ext_op_type == "gradient_based") {
+            mExtensionOperator = ExtensionOperator::GradientBased;
+        } else {
+            KRATOS_ERROR << "Wrong 'extension_operator_type' provided. Available options are 'MLS', 'RBF' and 'gradient_based'." << std::endl;
+        }
+
+        // Check that the basis settings are correct
+        KRATOS_ERROR_IF(mExtensionOperator == ExtensionOperator::RBF && !mConformingBasis)
+            << "'RBF' extension operator can only be used with conforming basis." << std::endl;
+        KRATOS_ERROR_IF(mExtensionOperator == ExtensionOperator::GradientBased && !mConformingBasis)
+            << "'gradient_based' extension operator can only be used with conforming basis." << std::endl;
+
+        // Set the SBD contion prototype to be used in the condition creation
+        std::string interface_condition_name = ThisParameters["sbm_interface_condition_name"].GetString();
+        KRATOS_ERROR_IF(interface_condition_name == "") << "SBM interface condition has not been provided." << std::endl;
+        mpConditionPrototype = &KratosComponents<Condition>::Get(interface_condition_name);
+    };
+
+    void ShiftedBoundaryMeshlessInterfaceUtility::CalculateExtensionOperator()
+    {
+        if (mExtensionOperator == ExtensionOperator::GradientBased) {
+            CalculateGradientBasedConformingExtensionBasis();
+        } else {
+            if (mConformingBasis) {
+                CalculateMeshlessBasedConformingExtensionBasis();
+            } else {
+                CalculateMeshlessBasedNonConformingExtensionBasis();
+            }
+        }
+    }
+
+    const Parameters ShiftedBoundaryMeshlessInterfaceUtility::GetDefaultParameters() const
+    {
+        const Parameters default_parameters = Parameters(R"({
+            "model_part_name" : "",
+            "boundary_sub_model_part_name" : "",
+            "sbm_interface_condition_name" : "",
+            "conforming_basis" : true,
+            "extension_operator_type" : "MLS",
+            "mls_extension_operator_order" : 1
+        })" );
+
+        return default_parameters;
+    }
+
+    void ShiftedBoundaryMeshlessInterfaceUtility::CalculateGradientBasedConformingExtensionBasis()

```

Comment:
longest function in the wild west

Seems that this could be shortened

Reply:
I'll try... But at some point I thought it will be easier to keep it like this.

Reply:
I went through it, the largest function is around 250 lines, which is admittedly larger than the standard rule of thumb of 200 lines. However, in this specific case, and taking into account that the implementation of the extension operators is a bit tricky, I think it would be better to leave it like this to ease following the steps one by one.